### PR TITLE
Card142/public figure2.0

### PR DESCRIPTION
**Pull request by dscherer21**

### CODE REVIEW

File: src/ProfilesPage/ProfilePage.js [hunk 1]
```diff
@@ -141,7 +159,7 @@ class ProfilePage extends Component {
   suspendUserButton() {
     const suspended = this.props.userInfo.suspended;
     const admin = sessionStorage.getItem("isAdmin");
-    if (this.state.suspendedState) {
+    /*if (this.state.suspendedState) {
```
Comment (vhoof) [inline src/ProfilesPage/ProfilePage.js]: Do we not want unsuspend buttons anymore?

File: src/ProfilesPage/ProfilePage.js [hunk 2]
```diff
@@ -193,15 +211,15 @@ class ProfilePage extends Component {
 
   renderPageContent() {
     const { userInfo, history } = this.props;
-    if (this.state.editUserEnable) {
+    /*if (this.state.editUserEnable) {
```
Comment (vhoof) [inline src/ProfilesPage/ProfilePage.js]: Do we not want this anymore?

File: src/actions/user_actions.js [hunk 3]
```diff
@@ -33,9 +34,9 @@ export const editUser = (boolean) => {
   }
 }
 
-export const updateSettings = (userId, adminStatus) => {
+export const updateSettings = (userId, adminStatus, callback) => {
   let switchAdmin;
-    adminStatus == true ? switchAdmin = false : switchAdmin = true;
+    switchAdmin =  adminStatus == true ? false : true;
```
Comment (vhoof) [inline src/actions/user_actions.js]: I think you could just delete both of these and use `{ "admin": !adminStatus }` on line 45 instead.

File: src/actions/user_actions.js [hunk 4]
```diff
@@ -46,10 +47,45 @@ export const updateSettings = (userId, adminStatus) => {
...
+
+export const updatePublicFigure = (userId, publicFigure, callback) => {
+ 
+  const request = axios({
+    method: 'patch',
+    url: process.env.REACT_APP_API_ENDPOINT + `users/${userId}`,
+    headers: {"id": userId },
+    data: { "public_figure":  !publicFigure }
+  });
+  console.log(publicFigure);
```
Comment (vhoof) [inline src/actions/user_actions.js]: Can probably remove this console log.

**Comment by vhoof** [comment]
I think you'll also need to merge `qa` in and fix some merge conflicts.

File: src/AdminPage/CreateAdmin.js [hunk 5]
```diff
@@ -5,17 +5,17 @@ import AdminForm from './components/AdminForm';
...
+import { fetchAllUsers, fetchUserInfo, updateSettings, updateEvent, updatePublicFigure, updateIsMentor, fetchNotifications } from '../actions';
 import './css/navbar.css';
 import EmailModal from './components/EmailModal';
 
 
 class CreateAdmin extends Component {
 
   componentDidMount() {
-    const { fetchAllUsers, fetchUserInfo, updateSettings, fetchNotifications } = this.props;
+    const { fetchAllUsers, fetchUserInfo, fetchNotifications } = this.props;
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/CreateAdmin.js]: done

File: src/AdminPage/CreateAdmin.js [hunk 6]
```diff
@@ -31,9 +31,16 @@ class CreateAdmin extends Component {
...
+          userInfo={this.props.userInfo} 
+          fetchAllUsers={this.props.fetchAllUsers} 
+          updatePublicFigure={this.props.updatePublicFigure} 
+          updateIsMentor={this.props.updateIsMentor} 
+          updateSettings={this.props.updateSettings} 
+          users={this.props.allUsersList} 
+          fetchUser={this.props.fetchUser}/>
         </PageContent>
-        <RightGraySideBar>
+        <RightGraySideBar>}
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/CreateAdmin.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 7]
```diff
@@ -43,14 +45,79 @@ class AdminForm extends Component {
     this.setState({ status: event.target.status });
   }
 
+  onClick(e, userId = this.props.userId) {
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 8]
```diff
@@ -43,14 +45,79 @@ class AdminForm extends Component {
...
+    e.preventDefault();
+    console.log('Button Clicked');
+    let publicFigureStatus = this.props.publicFigureStatus;
+    this.props.updatePublicFigure(userId, publicFigureStatus, this.props.fetchAllUsers);
+  }
+
+  handleClick(e, userId = this.props.userId) {
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 9]
```diff
@@ -43,14 +45,79 @@ class AdminForm extends Component {
     this.setState({ status: event.target.status });
   }
 
+  onClick(e, userId = this.props.userId) {
+    e.preventDefault();
+    console.log('Button Clicked');
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: done

File: src/ProfilesPage/ProfilePage.js [hunk 10]
```diff
@@ -77,11 +77,29 @@ class ProfilePage extends Component {
...
+    if (public_figure == true) {
+      return(
+        <h6>Public Figure</h6>
+      );
+    }
+  }
+
+  renderIsMentorStatus(userInfo = this.props.userInfo) {
+    let is_mentor = userInfo.is_mentor;
+    if (is_mentor == true) {
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: done

File: src/ProfilesPage/ProfilePage.js (hunk 1 above)
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: done

File: src/ProfilesPage/ProfilePage.js (hunk 2 above)
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 11]
```diff
@@ -43,14 +45,79 @@ class AdminForm extends Component {
...
+  }
+
+  renderIsMentor(event, userInfo = this.props.userInfo) {
+    /*if isMentor is true/false render according status*/
+    let isMentor = this.props.isMentorStatus;
+    let currentStatus;
+
+    isMentor === true
+      ? (currentStatus = "Mentor")
+      : (currentStatus = "Not a Mentor");
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 12]
```diff
@@ -66,11 +133,24 @@ class AdminForm extends Component {
       <div>
         <form onSubmit={this.handleSubmit}>
           <p id="admin-status-display">
-            Admin Status: {this.renderAdminStatus()}
+            Admin Status: {this.renderAdminStatus()}<br/><br/>
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: to get some spacing between each element.

File: src/AdminPage/components/UserList.js [hunk 13]
```diff
@@ -12,16 +12,22 @@ class UserList extends Component {
     const users = this.props.users.userList;
 
     return users.map(user => {
+      console.log(user);
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/UserList.js]: done

File: src/AdminPage/components/AdminForm.js [hunk 14]
```diff
@@ -43,14 +45,79 @@ class AdminForm extends Component {
...
+      )
+    } else {
+      return(
+        <button className="btn adminButtonStyle" onClick={this.handleClick}>Promote</button>
+      )
+    }
+  }
+
+  renderPublicFigure(event, userInfo = this.props.userInfo) {
+    /*if publicFigure is true/false render according status*/
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/AdminPage/components/AdminForm.js]: fixed

File: src/ProfilesPage/ProfilePage.js [hunk 15]
```diff
@@ -29,15 +29,15 @@ const admin = sessionStorage.getItem("isAdmin");
 // var editButtonClicked = false;
 
 class ProfilePage extends Component {
-  componentWillMount() {
+  componentDidMount() {
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: I thought componentWillMount() was bad practice?

File: src/ProfilesPage/ProfilePage.js [hunk 16]
```diff
@@ -77,11 +77,29 @@ class ProfilePage extends Component {
   renderEditUserButton() {
     return (
       <button className="btn btn-info" onClick={this.handleEditButtonClick.bind(this)}>
-        {this.state.editUserEnable ? "Back" : "Edit"}
+        {/*{this.state.editUserEnable ? "Back" : "Edit"}*/}
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: I think before it was causing something i was working on to glitch. fixed now.

File: src/ProfilesPage/ProfilePage.js [hunk 17]
```diff
@@ -77,11 +77,29 @@ class ProfilePage extends Component {
...
       </button>
     );
   }
 
+  renderPublicFigureStatus(userInfo = this.props.userInfo) {
+    let public_figure = userInfo.public_figure;
+    if (public_figure == true) {
```
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: done

File: src/ProfilesPage/ProfilePage.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (dscherer21) [reply src/ProfilesPage/ProfilePage.js]: @vhoof

File: src/ProfilesPage/ProfilePage.js [hunk 18]
```diff
@@ -103,6 +103,7 @@ class ProfilePage extends Component {
     // user is an admin or not. If it is the admin, render the Admin Edit form,
     // if it is a regular user, render user edit form.
     this.props.editUser(adminUser);
+    console.log(this.state.editUserEnable);
```
Comment (vhoof) [inline src/ProfilesPage/ProfilePage.js]: I'd remove this console.log

**Pull request merged**